pkg/e2e/operators/ocmagent/ocmagent.go

```diff
@@ -0,0 +1,235 @@
+	"github.com/openshift/osde2e/pkg/common/label"
+	"github.com/openshift/osde2e/pkg/e2e/operators"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+
+	operatorhubv1 "github.com/operator-framework/api/pkg/operators/v1alpha1"
+	monitoringv1 "github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring/v1"
+	appsv1 "k8s.io/api/apps/v1"
+	v1 "k8s.io/api/core/v1"
+	networkingv1 "k8s.io/api/networking/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	"sigs.k8s.io/e2e-framework/klient/k8s"
+	"sigs.k8s.io/e2e-framework/klient/k8s/resources"
+	"sigs.k8s.io/e2e-framework/klient/wait"
+	"sigs.k8s.io/e2e-framework/klient/wait/conditions"
+)
+
+var suiteName = "OCM Agent Operator"
+
+func init() {
+	alert.RegisterGinkgoAlert(suiteName, "SD_SREP", "@ocm-agent-operator", "sd-cicd-alerts", "sd-cicd@redhat.com", 4)
+}
+
+// TODO: Separate PR to remove 'Informing' label from test suite
+var _ = ginkgo.Describe(suiteName, ginkgo.Ordered, label.Operators, label.Informing, func() {
+	var h *helper.H
+	var client *resources.Resources
+
+	ginkgo.BeforeAll(func() {
+		h = helper.New()
+		client = h.AsUser("")
+	})
+
+	var (
+		configMapName      = "ocm-agent-config"
+		clusterRolePrefix  = "ocm-agent-operator"
+		deploymentName     = "ocm-agent"
+		namespace          = "openshift-ocm-agent-operator"
+		networkPolicyName  = "ocm-agent-allow-only-alertmanager"
+		operatorName       = "ocm-agent-operator"
+		operatorRegistry   = "ocm-agent-operator-registry"
+		secretName         = "ocm-access-token"
+		serviceMonitorName = "ocm-agent-metrics"
+		serviceName        = "ocm-agent"
+
+		deployments = []string{
+			deploymentName,
+			deploymentName + "-operator",
+		}
+	)
+
+	ginkgo.It("cluster service version exists", label.Install, func(ctx context.Context) {
+		var csvs operatorhubv1.ClusterServiceVersionList
+		err := client.List(ctx, &csvs, resources.WithFieldSelector(labels.FormatLabels(map[string]string{"metadata.namespace": namespace})))
+		expect.NoError(err)
+
+		for _, csv := range csvs.Items {
+			if csv.Spec.DisplayName == operatorName {
+				csv := &operatorhubv1.ClusterServiceVersion{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      csv.Name,
+						Namespace: namespace,
+					},
+				}
+				err = wait.For(conditions.New(client).ResourceMatch(
+					csv, func(object k8s.Object) bool {
+						obj := object.(*operatorhubv1.ClusterServiceVersion)
+						return obj.Status.Phase == "Succeeded"
+					}))
+				expect.NoError(err)
+				break
+			}
+		}
+	})
+
+	ginkgo.It("deployments exist", label.Install, func(ctx context.Context) {
+		for _, deploymentName := range deployments {
+			deployment := &appsv1.Deployment{}
+			err := client.Get(ctx, deploymentName, namespace, deployment)
+			expect.NoError(err)
+			Expect(deployment.Status.Replicas).To(BeNumerically("==", 1))
+			Expect(deployment.Status.ReadyReplicas).To(BeNumerically("==", 1))
+			Expect(deployment.Status.AvailableReplicas).To(BeNumerically("==", 1))
```

Comment:
Can we use the deployment available check from below here?

Reply:
Done!